hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestSparkSqlWithCustomKeyGenerator.scala

```diff
@@ -41,194 +42,208 @@ class TestSparkSqlWithCustomKeyGenerator extends HoodieSparkSqlTestBase {
-               | WHEN NOT MATCHED THEN INSERT (id, name, price, ts, segment)
-               |   values (source.id, source.name, source.price, source.ts, source.segment)
-               | """.stripMargin)
-          validateResults(
-            tableName,
-            s"SELECT id, name, cast(price as string), cast(ts as string), segment from $tableName",
-            tsGenFunc,
-            partitionGenFunc,
-            Seq(),
-            Seq(2, "a2", "11.9", 1704121827, "cat1"),
-            Seq(3, "a3", "30.0", 1706800227, "cat1"),
-            Seq(4, "a4", "113.4", 1701443427, "cat2"),
-            Seq(5, "a5", "2009.0", 1704121827, "cat2"),
-            Seq(6, "a6", "99.0", 1704121827, "cat3"),
-            Seq(7, "a7", "1409.0", 1706800227, "cat1"),
-            Seq(8, "a8", "24.9", 1706800227, "cat3"),
-            Seq(9, "a9", "299.0", 1701443427, "cat4"),
-            Seq(10, "a10", "888.8", 1706800227, "cat5")
-          )
-
-          // Test SQL DELETE
-          spark.sql(
-            s"""
-               | DELETE FROM $tableName
-               | WHERE id = 7
-               | """.stripMargin)
-          validateResults(
-            tableName,
-            s"SELECT id, name, cast(price as string), cast(ts as string), segment from $tableName",
-            tsGenFunc,
-            partitionGenFunc,
-            Seq(),
-            Seq(2, "a2", "11.9", 1704121827, "cat1"),
-            Seq(3, "a3", "30.0", 1706800227, "cat1"),
-            Seq(4, "a4", "113.4", 1701443427, "cat2"),
-            Seq(5, "a5", "2009.0", 1704121827, "cat2"),
-            Seq(6, "a6", "99.0", 1704121827, "cat3"),
-            Seq(8, "a8", "24.9", 1706800227, "cat3"),
-            Seq(9, "a9", "299.0", 1701443427, "cat4"),
-            Seq(10, "a10", "888.8", 1706800227, "cat5")
-          )
-
-          // Test DROP PARTITION
-          assertTrue(getSortedTablePartitions(tableName).contains(droppedPartition))
-          spark.sql(
-            s"""
-               | ALTER TABLE $tableName DROP PARTITION $dropPartitionStatement
-               |""".stripMargin)
-          validatePartitions(tableName, Seq(droppedPartition), expectedPartitions)
-
-          if (HoodieSparkUtils.isSpark3) {
-            // Test INSERT OVERWRITE, only supported in Spark 3.x
-            spark.sql(
-              s"""
-                 | INSERT OVERWRITE $tableName
-                 | SELECT 100 as id, 'a100' as name, 299.0 as price, 1706800227 as ts, 'cat10' as segment
-                 | """.stripMargin)
-            validateResults(
-              tableName,
-              s"SELECT id, name, cast(price as string), cast(ts as string), segment from $tableName",
-              tsGenFunc,
-              partitionGenFunc,
-              Seq(),
-              Seq(100, "a100", "299.0", 1706800227, "cat10")
-            )
+    for (extractPartition <- Seq(true, false)) {
+      withSQLConf(EXTRACT_PARTITION_VALUES_FROM_PARTITION_PATH.key() -> extractPartition.toString) {
```

Comment:
I think it's OK to use default value and deprecate `EXTRACT_PARTITION_VALUES_FROM_PARTITION_PATH` in the future.

Reply:
https://issues.apache.org/jira/browse/HUDI-8100